lib/message_customize/application_controller_patch.rb

```diff
@@ -12,15 +12,10 @@ def self.included(base)
     module InstanceMethod
       def reload_customize_messages
         custom_message_setting = CustomMessageSetting.find_or_default
-        return if custom_message_setting.latest_messages_applied?(current_user_language)
+        # NOTE: ApplicationController#set_localization sets the appropriate language in I18n.locale
+        return if custom_message_setting.latest_messages_applied?(I18n.locale)
 
-        MessageCustomize::Locale.reload!([current_user_language])
-      end
-
-      private
-
-      def current_user_language
-        User.current.language.presence || Setting.default_language
```

Comment:
This method is still being used in the following locations.
https://github.com/farend/redmine_message_customize/blob/f1761f3a74657960f596b3a68fbfabf9b60a6fb6/app/controllers/custom_message_settings_controller.rb#L57

As a result, when no message customization settings are registered, visiting the message customization settings screen (`/custom_message_settings/edit`) causes the following error:
```
NameError (undefined local variable or method `current_user_language' for an instance of CustomMessageSettingsController):
plugins/redmine_message_customize/app/controllers/custom_message_settings_controller.rb:57:in `set_lang'
```

How about fixing it like this?

```diff
diff --git a/lib/message_customize/application_controller_patch.rb b/lib/message_customize/application_controller_patch.rb
index 10b8abe..666210d 100644
--- a/lib/message_customize/application_controller_patch.rb
+++ b/lib/message_customize/application_controller_patch.rb
@@ -20,7 +20,7 @@ module MessageCustomize
       private
 
       def current_user_language
-        User.current.language.presence || Setting.default_language
+        I18n.locale
       end
     end
   end
```

Reply:
@hidakatsuya 
Thanks for the feedback. 

> How about fixing it like this?
> ...

You're right, that makes sense. I'll go with current_user_language as it's clearer and easier to maintain.
https://github.com/farend/redmine_message_customize/pull/68/commits/f8d6844440e607ace672cb36d7d8d5490e79ba3c fixes this problem.